docs/reactivity/overview.md

```diff
@@ -0,0 +1,240 @@
+
+This just a core concept, where you can optioanlly transform it into an `Ovee.js` component, add a way to destroy dialogs, to unregister them, etc., but it shows you that you're free to use reactive variables and watcher's in different scenarios.
+
+Let's take a look on one more example: global state management.
+
+## Building your own state managment system (store)
+
+In frameworks like `Vue` or `React`, there is often some sort of a global state management tool. In `React`, it's `Flux`, and in `Vue` it's `Vuex`. But you can make something much simpler for you small needs with little to none effort! Let's create an object that would store fetched posts.
+
+```js
+export const postsStore = makeReactive({
+    posts: [],
+    lastFetched: null,
+
+    getPostsByUserId: makeComputed(() => id => {
+        return this.posts.filter(post => post.user_id === id);
+    }),
+
+    readPosts: makeComputed(() => {
+        return this.posts.filter(post => post.status === 'read');
+    })
+})
+```
+
+Now we use `makeReactive` to create reactive object that can track changes of it's fields. Wi initialized `posts` field, to keep out posts and created to computeds: `getPostsByUserId`, that returns a function which than, based on passed id and state, returns different values, and `readPosts` that always returns only `read` posts.
+
+Now we can use it in our component, especially cool in `TemplateComponent`:
+
+```js
+import { postsStore } from '@/store'
+
+export class PostsList extends TemplateComponent {
+    template(html) {
+        return html`
+            <ul>
+                ${postsStore.readPosts.map(post => html`<li>${post.name}</li>`)}
+            </ul>
+        `;
+    }
+}
+```
+Note, that because we used `makeReactive`, we don't need to use `.value` when accesing `makeComputed` values.
+
+We can also make a list of all user-specific posts using `data-` attribute:
+
+```js
+import { postsStore } from '@/store'
+
+export class PostsList extends TemplateComponent {
+    @reactive()
+    @dataParam()
+    id;
+
+    template(html) {
+        return html`
+            <ul>
+                ${postsStore.getPostsByUserId(this.id).map(post => (
+                    html`<li>${post.name}</li>`
+                ))}
+            </ul>
+        `;
+    }
+}
+```
+
+Because `postsStore` is super simple and in no way protected, everybody can change it's state.
+
+To prevent that and minimalize bugs potential, you can restrict yourself and create special `postsMutations` or `postsActions` that only these methods will be allowed to change this state.
```

Comment:
```suggestion
To prevent that and minimalize the potential for bugs, you can restrict yourself and create special `postsMutations` or `postsActions` that only these methods will be allowed to change this state.
```

Reply:
Actually, you haven't shown the `postsMutations` and how to make the values protected from external writes ;)

Reply:
I think it would be too much work to show how to write system like that and I personally never thought about it xd

`postsMutations` was just an optional name for `postsActions` 😅 

I rewrited this paragraph a little 😉